src/lib/components/Assessment/index.js

```diff
@@ -0,0 +1,182 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {html} from "lit-element";
+import {BaseModalElement} from "../BaseModalElement";
+
+/**
+ * Element that renders a self-assessment callout.
+ * @extends {BaseModalElement}
+ * @final
+ */
+class Assessment extends BaseModalElement {
+  static get properties() {
+    return {
+      modal: {attribute: "aria-modal", reflect: true},
+    };
+  }
+
+  constructor() {
+    super();
+    this.modal = false;
+
+    this.onAssessmentAnimationEnd = this.onAssessmentAnimationEnd.bind(this);
+    this.onAssessmentResize = this.onAssessmentResize.bind(this);
+  }
+
+  render() {
+    if (!this.prerenderedChildren) {
+      this.prerenderedChildren = [];
+      this.setLeader = [];
+
+      for (const child of this.children) {
+        if (child.classList.contains("web-assessment__set-leader")) {
+          this.setLeader.push(child);
+        } else {
+          this.prerenderedChildren.push(child);
+        }
+      }
+    }
+
+    return html`
+      <div class="w-callout__header web-assessment__header">
+        <h2 class="w-callout__lockup web-assessment__lockup">
+          Check your understanding
+        </h2>
+        ${this.setLeader}
+        <button
+          @click="${this.onCloseClick}"
+          class="w-button--icon w-button--round web-assessment__close"
+          data-icon="close"
+        >
+          <span role="tooltip" class="w-tooltip">
+            Close
+          </span>
+        </button>
+      </div>
+      <button
+        @click="${this.onOpenClick}"
+        class="w-button w-button--primary web-assessment__open"
+      >
+        Open quiz
+      </button>
+      ${this.prerenderedChildren}
+    `;
+  }
+
+  firstUpdated() {
+    // Override BaseModalElement's inert behavior since Assessment opens itself.
+    this.inert = false;
+    this.classList.remove("unresolved");
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    const mqString = `(min-width: 481px)`;
+
+    matchMedia(mqString).addListener(this.onAssessmentResize);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    const mqString = `(min-width: 481px)`;
+
+    matchMedia(mqString).removeListener(this.onAssessmentResize);
+  }
+
+  onOpenClick() {
+    this.open = true;
+  }
+
+  onCloseClick() {
+    this.open = false;
+  }
+
+  updated(changedProps) {
+    if (changedProps.has("open")) {
+      this.modal = this.open;
+      if (this.open) {
+        this.openModal();
+      } else {
+        this.addEventListener("animationend", this.closeModal, {
```

Comment:
`closeModel` doesn't seem to be bound to `this`

Reply:
Woops